File: Version_Pruebas/ConexionPostgreSQL/Conectar-Insertar-Consultar-postgreSQL(archivo de prueba)/src/main/java/com/example/postgresql/daProjects.php


<?php

class daProjects {

    public function insert($objDemandantes) {
        $postgreSQL = new PostgreSQL();
        
        try {
            $con = $postgreSQL->connectDatabase();
            $stmt = $con->prepare("INSERT INTO developer.demandante (id_demandante, nombre, apellido, edad) VALUES (?, ?, ?, ?)");

            foreach ($objDemandantes as $demandante) {
                $stmt->bindParam(1, $demandante->getid());
                $stmt->bindParam(2, $demandante->getNombre());
                $stmt->bindParam(3, $demandante->getApellido());
                $stmt->bindParam(4, $demandante->getEdad());
                $stmt->execute();
            }
        } catch (PDOException $ex) {
            echo $ex->getMessage();
        }
    }

    public function select() {
        $postgreSQL = new PostgreSQL();
        $query = "SELECT * FROM developer.demandante";

        try {
            $con = $postgreSQL->connectDatabase();
            $stmt = $con->prepare($query);
            $stmt->execute();

            while ($row = $stmt->fetch(PDO::FETCH_ASSOC)) {
                echo "project_id: " . $row['id_demandante'] . " | ";
                echo "name: " . $row['nombre'] . " | ";
                echo "created_at: " . $row['apellido'] . " | ";
                echo "project_edad: " . $row['edad'] . " | ";
                echo "<br>";
            }
        } catch (PDOException $ex) {
            echo $ex->getMessage();
        }
    }

    public function delete($id_demandante) {
        $postgreSQL = new PostgreSQL();
        $query = "DELETE FROM developer.demandante WHERE id_demandante = :id_demandante";

        try {
            $con = $postgreSQL->connectDatabase();
            $stmt = $con->prepare($query);
            $stmt->bindParam(':id_demandante', $id_demandante);
            $stmt->execute();
            echo "Registro eliminado exitosamente.";
        } catch (PDOException $ex) {
            echo "Error al eliminar el registro: " . $ex->getMessage();
        }
    }

    public function update($id_demandante, $nombre, $apellido, $edad) {
        $postgreSQL = new PostgreSQL();
        $query = "UPDATE developer.demandante SET nombre = :nombre, apellido = :apellido, edad = :edad WHERE id_demandante = :id_demandante";

        try {
            $con = $postgreSQL->connectDatabase();
            $stmt = $con->prepare($query);
            $stmt->bindParam(':nombre', $nombre);
            $stmt->bindParam(':apellido', $apellido);
            $stmt->bindParam(':edad', $edad);
            $stmt->bindParam(':id_demandante', $id_demandante);
            $stmt->execute();
            echo "Registro actualizado exitosamente.";
        } catch (PDOException $ex) {
            echo "Error al actualizar el registro: " . $ex->getMessage();
        }
    }
}

?>

File: Version_Pruebas/ConexionPostgreSQL/Conectar-Insertar-Consultar-postgreSQL(archivo de prueba)/src/main/java/com/example/postgresql/daProjects.java
package com.example.postgresql;

import java.sql.*;

public class daProjects {

    public void insert(Demandante[] objDemandantes) throws SQLException {
        PostgreSQL postgreSQL = new PostgreSQL();
        // Demandante objDemandantes = new Demandante();
        // String query = String.format("INSERT INTO
        // developer.projects(name,description) VALUES('%s','%s')",projects.getName(),
        // projects.getDescription());
        // String sql = "INSERT INTO developer.demandante (id_demandante, nombre,
        // apellido, edad) VALUES('" + objDemandantes.getid() + "', '" +
        // objDemandantes.getNombre() + "', '" + objDemandantes.getApellido() + "', '" +
        // objDemandantes.getedad() + "')";

        try {
            Connection con = postgreSQL.connectDatabase();
            Statement stmnt = null;
            stmnt = con.createStatement();
            for (Demandante demandate : objDemandantes) {
                String sql = "INSERT INTO developer.demandante (id_demandante, nombre, apellido, edad) VALUES('"
                        + demandate.getid() + "', '" + demandate.getNombre() + "', '" + demandate.getApellido() + "', '"
                        + demandate.getedad() + "')";

                // stmnt.executeUpdate(query);
                stmnt.executeUpdate(sql);
            }
        } catch (SQLException ex) {
            System.out.println(ex.getMessage());
        }
    }

    public void select() {
        PostgreSQL postgreSQL = new PostgreSQL();
        String query = "SELECT * FROM developer.demandante";

        try (Connection con = postgreSQL.connectDatabase();
                PreparedStatement pst = con.prepareStatement(query);
                ResultSet rs = pst.executeQuery()) {

            while (rs.next()) {

                System.out.print("project_id: " + rs.getString(1));
                System.out.print(" | ");
                System.out.print("name: " + rs.getString(2));
                System.out.print(" | ");
                System.out.println("created_at: " + rs.getString(3));
                System.out.print(" | ");
                System.out.print("project_edad " + rs.getString(4));
                System.out.print(" | ");
            }

        } catch (SQLException ex) {
            System.out.println(ex.getMessage());
        }
    }

    public void delete(int id_demandante) {
        PostgreSQL postgreSQL = new PostgreSQL();
        String query = "DELETE FROM developer.demandante WHERE id_demandante = " + id_demandante;

        try (Connection con = postgreSQL.connectDatabase();
                Statement stmnt = con.createStatement()) {
            stmnt.executeUpdate(query);
            System.out.println("Registro eliminado exitosamente.");
        } catch (SQLException ex) {
            System.out.println("Error al eliminar el registro: " + ex.getMessage());
        }
    }

    public void update(int id_demandante, String nombre, String apellido, int edad) {
        PostgreSQL postgreSQL = new PostgreSQL();
        String query = "UPDATE developer.demandante SET nombre = '" + nombre + "', apellido = '" + apellido +
                "', edad = " + edad + " WHERE id_demandante = " + id_demandante;

        try (Connection con = postgreSQL.connectDatabase();
                Statement stmnt = con.createStatement()) {
            stmnt.executeUpdate(query);
            System.out.println("Registro actualizado exitosamente.");
        } catch (SQLException ex) {
            System.out.println("Error al actualizar el registro: " + ex.getMessage());
        }
    }
}
